src/components/HungerAlert/HungerAlert.tsx

```diff
@@ -0,0 +1,64 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import PopupModal from '@/components/PopupModal/PopupModal';
+import DataTable from '@/components/DataTable/DataTable';
+import './style.css';
+import container from '@/container';
+import { GlobalDataRepository } from '@/domain/repositories/GlobalDataRepository';
+
+export default function HungerAlert() {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [countriesWithHighHunger, setCountriesWithHighHunger] = useState<
+    { rank: number; country: string; fcs: string }[]
+  >([]);
+
+  const toggleModal = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+
+  useEffect(() => {
+    const fetchCountriesWithHighHunger = async () => {
+      const countryData = await container
+        .resolve<GlobalDataRepository>('GlobalDataRepository')
+        .getMapDataForCountries();
```

Comment:
you shouldn't download this data here, because this is already downloaded on the server in `page.tsx`. You can pass that data to this component. However, you would have to move this component to the page.tsx file, but I don't think that's a problem, since we will only have one page. Or you could fetch the data in the `layout.tsx` as well, I think next.js is smart enough not to make the same call twice.

Reply:
The problem with this endpoint that it returns this error when its called from server side componets  =>
 **{
    "status": "resolved_model",
    "value": "{\"error\":\"Error retrieving data: Error fetching ADM0 data: ('Connection broken: IncompleteRead(1604 bytes read, 8636 more expected)', IncompleteRead(1604 bytes read, 8636 more expected))\"}",
    "reason": null,
    "_response": {
        "_bundlerConfig": null,
        "_moduleLoading": null,
        "_chunks": {},
        "_stringDecoder": {},
        "_rowState": 0,
        "_rowID": 0,
        "_rowTag": 0,
        "_rowLength": 0,
        "_buffer": []
    },
    "_debugInfo": null
}**
but when its called from client side component this error never occures. I have tried different endpoints on server side component and this issue never happens , I am not sure how exactly next.js handles those requests on different rendering components. so the work around that I have done is to pass data from layout.js (which always returns the above error) if it returns error just call the endpoint again from the client side which eventually will be resolved without an error.

Reply:
oh I think this happens because a few days ago, their API was broken and this response was returned, and next cached it locally. When you make the request now, it returns the correct data, but it's over 2 MB so it can't update the cache with that. You can fix this by deleting the .next folder in the root dir, and then running yarn dev again.